### PR TITLE
fix(CaptureMatching) - crashes with Invalid time value #836

### LINKED ISSUE #836: Capture Matching tool crashes with "Invalid time value"
Screenshot 2022-09-19 at 20 57 43

### PR DESCRIPTION
**Pull request by Vlad-Pischaeff**
## Description

Convert string values to new Date() for format

**Issue(s) addressed**

- Resolves #836

**What kind of change(s) does this PR introduce?**

- [ ] Enhancement
- [x] Bug fix
- [ ] Refactor

**Please check if the PR fulfils these requirements**

- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## Issue

**What is the current behavior?**
CaptureMatching tool crashes with Invalid time value

**What is the new behavior?**
Right date/time values show

## Breaking change

**Does this PR introduce a breaking change?**
No

### CODE REVIEW

File: src/components/CaptureMatching/CaptureMatchingView.js [hunk 1]
```diff
@@ -424,6 +424,13 @@ function CaptureMatchingView() {
     [captureImage?.latitude, captureImage?.longitude]
   );
 
+  const getGroverRegDate = () => {
```
Comment (nmcharlton) [inline src/components/CaptureMatching/CaptureMatchingView.js:427]: Typo: `Grover` -> `Grower`

File: src/components/CaptureMatching/CaptureMatchingView.js [hunk 2]
```diff
@@ -424,6 +424,13 @@ function CaptureMatchingView() {
...
+    return growerAccount.first_registration_at
+      ? 'Joined at ' +
+          format(new Date(growerAccount.first_registration_at), 'MM/dd/yyyy')
```
Comment (nmcharlton) [inline src/components/CaptureMatching/CaptureMatchingView.js:430]: Can we update this to use `getDateTimeStringLocale()` so we use the appropriate format for the user?

**Review by nmcharlton — CHANGES_REQUESTED** [review]

File: src/components/CaptureMatching/CaptureMatchingView.js (hunk 2 above)
Comment (Vlad-Pischaeff) [reply src/components/CaptureMatching/CaptureMatchingView.js:430]: I think so

File: src/components/CaptureMatching/CaptureMatchingView.js [hunk 3]
```diff
@@ -427,8 +426,8 @@ function CaptureMatchingView() {
   const getGroverRegDate = () => {
     return growerAccount.first_registration_at
       ? 'Joined at ' +
-          format(new Date(growerAccount.first_registration_at), 'MM/dd/yyyy')
-      : 'Joined at ' + format(new Date(growerAccount.created_at), 'MM/dd/yyyy');
+          getDateFormatted(growerAccount.first_registration_at, 'dd.MM.yyyy')
```
Comment (nmcharlton) [inline src/components/CaptureMatching/CaptureMatchingView.js:429]: The intention is to format this date according to the user's locale-language (e.g., `mm/dd/yyyy` for `en-US`, `yyyy/mm/dd` for `zh-ZH`) using [`Intl.DateTimeFormat`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat).

`getDateTimeStringLocale()` already does the formatting we want, so it'd be better to use or update that than create a new method.

**Review by nmcharlton — CHANGES_REQUESTED** [review]

**Review by nmcharlton — APPROVED** [review]

**Pull request merged**